### PR TITLE
[WIP] nox wrappers added

### PR DESCRIPTION
**Pull request by andreamola**
first version of trilinos nox nonlinear solver equation wrapper added.
the wrapper is implemented in class NewtonSolver and makes used of the NewtonArgument virtual class.
nox_wrapper.cc has been added in the tests directory to test the wrapper.

### CODE REVIEW

**Comment by luca-heltai** [comment]
Andrea, puoi mettere un po' di `#ifdef DEAL_II_WITH_TRILINOS` per permettere alla libreria di compilare anche se trilinos non è installata?

Ricordati anche di far girare lo script di indentazione, che ora Travis si lamenta anche dell'indentazione...

**Comment by luca-heltai** [comment]
Travis is still not happy... 

https://travis-ci.org/luca-heltai/dealii-sak/builds/65410094#L391

I think also the cc files need some ifdefs...

**Comment by luca-heltai** [comment]
Ah, no. This is the problem:

rename the file 

`tests/nox_wrapper.output`

to 

`tests/nox_wrapper.with_trilinos=on.output`

File: include/newton_argument.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#ifndef newton_argument_h
+#define newton_argument_h
+
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/vector_view.h>
+
+using namespace dealii;
+
+#ifdef DEAL_II_WITH_TRILINOS
```
Comment (luca-heltai) [inline include/newton_argument.h]: In this class you never use anything from Trilinos. You only declare an interface, correct? Then why exclude it if one does not have trilinos? Isn't this the same interface (more or less) of the ida solver?

**Comment by andreamola** [comment]
Ok fixed the travis problem. I'm off for ibiza tomorrow morning and will be back in italy on monday night. I think the laptop is staying here, but I can be reached via email or phone.
Btw, check nicola. He has good news too ;)

**Comment by luca-heltai** [comment]
Ok,  I made NC clean up the code a bit, and take it over to remove some commented code. Nice Job!

**Pull request closed without merging**